dask_kubernetes/operator/controller/controller.py

```diff
@@ -632,24 +638,65 @@ async def daskworkergroup_replica_update(
                 if "labels" in worker_spec["metadata"]:
                     labels.update(**worker_spec["metadata"]["labels"])
             if workers_needed > 0:
-                for _ in range(workers_needed):
-                    data = build_worker_deployment_spec(
-                        worker_group_name=name,
-                        namespace=namespace,
-                        cluster_name=cluster_name,
-                        uuid=uuid4().hex[:10],
-                        pod_spec=worker_spec["spec"],
-                        annotations=annotations,
-                        labels=labels,
-                    )
-                    kopf.adopt(data, owner=body)
-                    kopf.label(data, labels=cluster_labels)
-                    await kubernetes.client.AppsV1Api(
-                        api_client
-                    ).create_namespaced_deployment(
-                        namespace=namespace,
-                        body=data,
-                    )
+                if deployment_groups:
+                    for __ in range(deployment_groups):
+                        for _ in range(SIZE):
+                            data = build_worker_deployment_spec(
+                                worker_group_name=name,
+                                namespace=namespace,
+                                cluster_name=cluster_name,
+                                uuid=uuid4().hex[:10],
+                                pod_spec=worker_spec["spec"],
+                                annotations=annotations,
+                                labels=labels,
+                            )
+                            kopf.adopt(data, owner=body)
+                            kopf.label(data, labels=cluster_labels)
+                            await kubernetes.client.AppsV1Api(
+                                api_client
+                            ).create_namespaced_deployment(
+                                namespace=namespace,
+                                body=data,
+                            )
+                        await asyncio.sleep(DELAY)
```

Comment:
Instead of having this concept of deployment groups and batching things like this, we could do 

```python
for _ in range(min(workers_needed, SIZE)):
``` 

and then if we don't have enough workers we could 

```python
if workers_needed > SIZE:
    raise kopf.TemporaryError(
        "Add maximum number of workers for this batch but still need to create more workers, " 
        f"waiting for {DELAY} seconds before continuing.", 
        delay=DELAY)
```

This will cause `kopf` to stop executing `daskworkergroup_replica_update`, wait for the delay period and then call it again.

https://kopf.readthedocs.io/en/stable/errors/#temporary-errors

I think this would simplify things a lot here. We just let `kopf` keep calling `daskworkergroup_replica_update` until we have all the workers we requested. It also means if the autoscaler reduces the number of workers in the meantime we will stop scaling up.